src/mon/MDSMonitor.cc

```diff
@@ -985,15 +986,19 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
     }
     set<mds_gid_t> stopping;
     for (auto& id : ids) {
-      ostringstream ess;
       mds_gid_t gid = gid_from_arg(fsmap, id, ess);
```

Comment:
this function doesn't print newlines in the `ess`. You should inject a new line in this string every time you get MDS_GID_NONE to delimit the different error messages.